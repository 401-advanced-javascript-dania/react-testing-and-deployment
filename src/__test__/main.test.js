import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Main from '../modules/main.js';
describe('<Main/>',()=>{
    it('exists at the start of our app',()=>{
        let app =shallow(<Main/>);
        expect(app.find('h2').exists()).toBeTruthy();
    })
    it('increament the number on click',()=>{
        let app = mount(<Main/>);
        let button=app.find('.one');
        button.simulate('click');
        expect(app.state('counter')).toEqual(1);
    })
    it('render correctly',() =>{
        const tree = renderer.create(<Main/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})