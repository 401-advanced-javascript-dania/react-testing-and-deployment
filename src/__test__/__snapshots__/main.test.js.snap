// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> render correctly 1`] = `
<article>
  <h2>
    0
  </h2>
  <button
    className="one"
    onClick={[Function]}
  >
    counterup
  </button>
  <button
    onClick={[Function]}
  >
    counterdown
  </button>
</article>
`;
